Engine/media/audio/audio.cpp

```diff
@@ -918,41 +998,46 @@ void update_music_volume() {
                 newvol = targetVol;
                 stop_and_destroy_channel_ex(SCHAN_MUSIC, false);
                 if (crossFading > 0) {
-                    channels[SCHAN_MUSIC] = channels[crossFading];
-                    channels[crossFading] = nullptr;
+                    lock.SetChannel(SCHAN_MUSIC, lock.GetChannel(crossFading));
+                    lock.SetChannel(crossFading, nullptr);
                 }
                 crossFading = 0;
             }
             else {
                 if (crossFading > 0)
-                    channels[crossFading]->set_volume((curvol > targetVol) ? targetVol : curvol);
+                    lock.GetChannel(crossFading)->set_volume((curvol > targetVol) ? targetVol : curvol);
```

Comment:
This grabs the channel without checking for null.   I only just noticed a couple of instance of these.. haven't thoroughly checked for others.

Reply:
Here, and in other places, engine is relying on other variables like `crossFading`.

Reply:
Yes but the crossFading variable is not protected by the channel locks.

Reply:
> Yes but the crossFading variable is not protected by the channel locks.

This is true...